### PR TITLE
Update resource section

### LINKED ISSUE #2414: Add resource section in the documentation
What do you think about adding a section in the docs with links to and short descriptions of learning resources and packages related to / building on Altair?

I went through all Github repos mentioning Altair that was updated since Aug 2020 and labeled as either a Python and Notebook repo. From that search, the resource section could look something like this:

---

# Resources

## Learning material

- [Jupyter Notebooks tutorials and examples](https://github.com/altair-viz/altair_notebooks)
- [A data visualization curriculum from the UW data group](https://uwdata.github.io/visualization-curriculum)
- [Altair tutorial given at PyCon 2018](https://altair-viz.github.io/altair-tutorial)

## Related packages

- [Vega-Lite - The higher-level visualization grammar that Altair implements in Python](https://vega.github.io/vega-lite/)
- [altair_saver - Enables saving charts to a variety of output types](https://github.com/altair-viz/altair_saver)
- [altair_data_server - Data transformer plugin that transparently serves data for charts](https://github.com/altair-viz/altair_data_server)
- [altair_pandas - Altair backend for the pandas plotting API](https://github.com/altair-viz/altair_pandas)
- [vega_datasets -  Offline access to the Vega datasets used in the Altair documentation](https://github.com/altair-viz/vega_datasets)
- [nx_altair - Draw interactive NetworkX graphs with Altair](https://github.com/Zsailer/nx_altair)
- [gif - Create animated Altair gifs via loops](https://github.com/maxhumber/gif)
- [Vegawidget - R interface to Altair](https://vegawidget.github.io/altair/)

---

For full disclosure, I am working on a data visualization course using Altair at the University of British Columbia and a wrapper package for making a few EDA plots quicker with Altair. When they are further along, I will suggest adding them here, but also happy if this stays as is and just makes the info easier to find.

### PR DESCRIPTION
**Pull request by joelostblom**
Somehow I missed that this section already existed when opening #2414. It was quite out of date, so I removed packages that haven't seen updates in a few years and added new ones as well a section on the learning resources. I also attempted to make this more discoverable by adding links from the Overview and Gallery pages.

close #2414

### CODE REVIEW

**Comment by joelostblom** [comment]
Not sure if the doctest will serve a live preview, but in case it doesn't, here is screenshot of what the section looks like now:

image

**Comment by harabat** [comment]
@joelostblom 
There are also two dashboarding tools, [Panel](https://github.com/holoviz/panel) and [Streamlit](https://github.com/streamlit/streamlit), that might be relevant. While they also support other plotting libraries, Altair features pretty heavily in their communication and communities. Would that be part of the ecosystem?

**Comment by joelostblom** [comment]
Thanks @harabat , that's a good point you bring up. I was personally thinking about this section as tools that are more directly related to Altair rather than just compatible with it. But we do mention `gif` which is a bit more general, and [the Altair dashboard question on Stack Overflow](https://stackoverflow.com/questions/49833866/making-dashboards-using-altair) is quite popular so it would make sense to say something about Dashboards I think...

If streamlit starts integrating specific Altair functionality, that might be the natural tool to highlight, but for the time being, I think I will write up a more elaborate example to the SO question with an example from panel, dash and streamlit and then let @jakevdp decide if he thinks that section should be copied in the docs also or maybe just link to it somewhere and mention that Altair is compatible with these tools.

I actually opened a related issue in the dash docs regarding the preferred way to use other plotting libraries, but haven't heard back yet https://github.com/plotly/dash-docs/issues/1068

**Comment by jcmkk3** [comment]
There's also the video series about Altair from [calmcode](https://calmcode.io/altair/introduction.html).

**Comment by firasm** [comment]
@jcmkk3 - wow Thank you! I wish I had seen those videos a few weeks ago! Definitely will incorporate them as part of my class next time I teach Altair!

**Comment by joelostblom** [comment]
@jakevdp I added a reference to the course I mentioned in the original issue and to the [EDA wrapper package I made](https://joelostblom.github.io/altair_ally/intro.html) (let me know if you think logo or name are too official-looking and I'm happy to change). The failing test is related to a file that I haven't touched in this PR (altair/sphinxext/schematable.py). This is not the most pressing PR, I know you're very short on time.

@harabat I added [a more elaborated reply to the dashboard SO question](https://stackoverflow.com/questions/49833866/making-dashboards-using-altair) and linked it from the resource section.

@jcmkk3 @firasm I linked to the calmcode videos as well.

**Comment by firasm** [comment]
Thanks @joelostblom !

What you have is great! I just want to add a couple of things that @eitanlees created that I found super helpful:

- [What is Altair](https://www.youtube.com/watch?v=AAuPPorsmJc)
- [Altair Visualization Grammar](https://www.youtube.com/watch?v=U7w1XumKK60) and [accompanying slides](https://eitanlees.com/Visualization-Grammar.pdf)
- [Altair Data Types video](https://www.youtube.com/watch?v=LSEPyCqjoAg)
- [Understanding the Altair stack](https://eitanlees.github.io/altair-stack/)

**Comment by joelostblom** [comment]
Thanks @firasm, I agree those are great and @eitanlees already have an open PR https://github.com/altair-viz/altair/pull/2292 to include many of them in the main user guide which I think is even better than having them here.

File: doc/user_guide/ecosystem.rst [hunk 1]
```diff
@@ -1,20 +1,119 @@
...
+
+Vega-Lite_
+~~~~~~~~~~
+
+The higher-level visualization grammar that Altair implements in Python.
+
+.. List of links.
+.. _Vega-Lite: https://vega.github.io/vega-lite
+
+altair_saver_
```
Comment (binste) [inline doc/user_guide/ecosystem.rst:75]:
```suggestion
.. _Vega-Lite: https://vega.github.io/vega-lite

vl-convert_
~~~~~~~~~~~

Python library for converting Altair/Vega-Lite chart specifications into static images (SVG or PNG) or Vega chart specifications without any external dependencies.

.. List of links.
.. _vl-convert: https://github.com/vega/vl-convert

altair_saver_
```

@joelostblom Happy to review this PR if you want

File: doc/user_guide/ecosystem.rst [hunk 2]
```diff
@@ -1,20 +1,127 @@
...
+
+vl-convert_
+~~~~~~~~~~~
 
-This is a list of projects related to or built on top of Altair. We hope to make it easier to find these projects by listing them here. If you know of a project that should be added, please let us know.
+Python library for converting Altair/Vega-Lite chart specifications into static images (SVG or PNG) or Vega chart specifications without any external dependencies.
+
+.. List of links.
+.. _vl-convert: https://github.com/vega/vl-convert
+
```
Comment (joelostblom) [inline doc/user_guide/ecosystem.rst:82]:
```suggestion

VegaFusion_
~~~~~~~~~~~

VegaFusion provides server-side scaling for Altair charts, which can accelerate interactive charts, extract transformed data, and perform data-intensive aggregations on the server and prune unused columns from the source datase yielding smaller size visualizations

.. List of links.
.. _VegaFusion: https://vegafusion.io/

```

File: doc/user_guide/ecosystem.rst (hunk 1 above)
Comment (joelostblom) [reply doc/user_guide/ecosystem.rst]: @binste Yes that would be great, thank you!

File: doc/user_guide/ecosystem.rst [hunk 3]
```diff
@@ -1,20 +1,131 @@
...
+Jupyter Notebooks tutorials and examples from the Altair authors.
+
+.. List of links.
+.. _`Jupyter Notebook Tutorials`: https://github.com/altair-viz/altair_notebooks
+
+`Pycon Tutorial`_
+~~~~~~~~~~~~~~~~~
+
+Altair tutorial given at PyCon 2018 by the Altair authors.
```
Comment (binste) [inline doc/user_guide/ecosystem.rst:34]:
```suggestion
Altair tutorial given at PyCon 2018 by the Altair author Jake VanderPlas.
```
Maybe make it more specific as the recording shows only Jake giving that talk?

File: doc/user_guide/ecosystem.rst [hunk 4]
```diff
@@ -1,20 +1,131 @@
...
+The higher-level visualization grammar that Altair implements in Python.
+
+.. List of links.
+.. _Vega-Lite: https://vega.github.io/vega-lite
+
+vl-convert_
+~~~~~~~~~~~
+
+
+VegaFusion_
```
Comment (binste) [inline doc/user_guide/ecosystem.rst:79]: Somehow my previous code suggestion wasn't applied completely, only the title, so I pasted it again below.
```suggestion
vl-convert_
~~~~~~~~~~~

Python library for converting Altair/Vega-Lite chart specifications into static images (SVG or PNG) or Vega chart specifications without any external dependencies.

.. List of links.
.. _vl-convert: https://github.com/vega/vl-convert

VegaFusion_
```

File: doc/user_guide/ecosystem.rst [hunk 5]
```diff
@@ -1,20 +1,131 @@
...
----------------
+~~~~~~~~~~~~~~~
 
-altair-recipes provides a collection of ready-made statistical graphics for vega. See the `docs <https://altair-recipes.readthedocs.io/en/latest/>`__.
+altair_recipes provides a collection of ready-made statistical graphics for Altair. See the `docs <https://altair-recipes.readthedocs.io/en/latest/>`__.
 
 .. List of links.
 .. _altair_recipes: https://github.com/piccolbo/altair_recipes
 
 gpdvega_
```
Comment (binste) [inline doc/user_guide/ecosystem.rst:127]: Is this package still of use? I'm not familiar with it so maybe I miss something. Last commit on GitHub is from 2018 and it's main feature seems to be support for GeoPandas dataframes which was added to Altair in 3.3.0 (2019). I was able to execute the examples from the documentation of gpdvega without having it installed or importing it.

Can't make a code suggestion here as the diff range is too short.

**Review by binste — CHANGES_REQUESTED** [review]

File: doc/user_guide/ecosystem.rst (hunk 5 above)
Comment (joelostblom) [reply doc/user_guide/ecosystem.rst:127]: Good point, I think we can remove it since the Altair docs now nicely shows how to work with geodata

Comment (joelostblom) [reply doc/user_guide/ecosystem.rst]: Removed

**Comment by joelostblom** [comment]
Thanks for the comments @binste ! I accepted all your suggestions and shortened the title a bit; merging this

**Pull request merged**